test/extensions/filters/http/local_ratelimit/local_ratelimit_integration_test.cc

```diff
@@ -489,6 +519,46 @@ TEST_P(LocalRateLimitFilterIntegrationTest, DenyRequestWithinSameConnection) {
   EXPECT_EQ(18, response->body().size());
 }
 
+TEST_P(LocalRateLimitFilterIntegrationTest, LimitHeaderTest) {
```

Comment:
Could you also include additional tests to test the header presence in various scenarios including partial fills? Using the `1000s` in interval makes it hard to verify the decreasing behavior.